core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -1526,6 +1526,37 @@ public void desyncDocumentFromRemote(
     triggerListeningToNamespace(namespace);
   }
 
+  /**
+   * A document that is paused no longer has remote updates applied to it.
+   * Any local updates to this document cause it to be resumed. An example of pausing a document
+   * is when a conflict is being resolved for that document and the handler throws an exception.
+   *
+   * This method allows you to resume sync for a document.
+   *
+   * @param documentId the id of the document to resume syncing
+   * @return true if successfully resumed, false if the document
+   *         could not be found or there was an error resuming
+   */
+  boolean resumeSyncForDocument(
+      final MongoNamespace namespace,
+      final BsonValue documentId
+  ) {
+    if (namespace == null || documentId == null) {
+      return false;
+    }
+
+    final NamespaceSynchronizationConfig namespaceSynchronizationConfig;
+    final CoreDocumentSynchronizationConfig config;
+
+    if ((namespaceSynchronizationConfig = syncConfig.getNamespaceConfig(namespace)) == null
+        || (config = namespaceSynchronizationConfig.getSynchronizedDocument(documentId)) == null) {
+      return false;
+    }
+
+    config.setPaused(false);
+    return !config.isPaused();
```

Comment:
won't this always be true?

Reply:
Thought there's a small chance of a race here– document is resumed and in that instant an error occurs again. Returning !isPaused seemed like a sure way to return the state of the doc. I could leave a TODO for your threading ticket. Or I could be being overly careful here.

Reply:
I feel like it's fine to return true. The same thing can happen even right after it returns to you.